source/Calamari/Deployment/PackageRetention/Caching/FirstInFirstOutCacheAlgorithm.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Calamari.Common.Plumbing.Deployment.PackageRetention;
+using Calamari.Deployment.PackageRetention.Model;
+
+namespace Calamari.Deployment.PackageRetention.Caching
+{
+    public class FirstInFirstOutCacheAlgorithm : RetentionAlgorithmBase
```

Comment:
What's the purpose of this cache algorithm?

Reply:
This was originally created to give James something to work off when implementing his part of the process.  I've left it in here for now as a fall-back algorithm in case we have issues with the more involved LFU algorithm.

Good pick up!